lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -2377,6 +2380,19 @@ def validate_alert_rules(self, rules: dict) -> Tuple[bool, str]:
                     ]
                 )
 
+    def validate_scrape_jobs(self, jobs):
+        """Validate scrape jobs using cos-tool."""
+        conf = {"scrape_configs": jobs}
+        with tempfile.NamedTemporaryFile() as tmpfile:
+            with open(tmpfile.name, "w") as f:
+                f.write(yaml.safe_dump(conf))
+            try:
+                self._exec([str(self.path), "validate-config", tmpfile.name])
+            except subprocess.CalledProcessError as e:
+                logger.error("Validating scrape jobs failed: {}".format(e.output))
+                return False
```

Comment:
1. I wonder if this should raise instead
2. (Haven't looked at the context) what if cos-tool isn't present? Do we lose all scrape jobs?

Reply:
Yes we would lose all scrape jobs. Fixing now.

Reply:
fixed

Reply:
We don't raise in other places that we [do validation either](https://github.com/canonical/prometheus-k8s-operator/blob/main/lib/charms/prometheus_k8s/v0/prometheus_scrape.py#L2381). Whether or not we should/could raise is a question, but raising an exception back to be caught... wherever is simply re-shuffling the logic in the caller from `if/else` to `try/except` and moving the `logger.error` here to there, for an exception which nothing can really meaningfully be done about it anyway.

What advantages would raising have?

Reply:
I don't think raising makes sense here. We certainly don't want to enter error state. Avoiding that is the whole point of this change. Raising and catching it elsewhere, as Ryan said, more or less doesn't do anything.